src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -626,6 +625,56 @@ public byte[] Http2AltSvcOriginUri
             return await GetHttpConnectionAsync(request, cancellationToken).ConfigureAwait(false);
         }
 
+        private Http2Connection? GetHttp2ConnectionAcceptingNewStreams()
```

Comment:
Is it possible for two threads to call into this method at the same time, both return null, and both open separate new connections?

Reply:
Never mind, `_http2ConnectionCreateLock` prevents this.